PWGHF/HFC/TableProducer/correlatorDplusHadrons.cxx

```diff
@@ -331,19 +453,81 @@ struct HfCorrelatorDplusHadrons {
         if ((std::abs(particle2.pdgCode()) != 11) && (std::abs(particle2.pdgCode()) != 13) && (std::abs(particle2.pdgCode()) != 211) && (std::abs(particle2.pdgCode()) != 321) && (std::abs(particle2.pdgCode()) != 2212)) {
           continue;
         }
+        int poolBin = corrBinningMCGen.getBin(std::make_tuple(mccollision.posZ(), getTracksSize(mccollision)));
         entryDplusHadronPair(getDeltaPhi(particle2.phi(), particle1.phi()),
                              particle2.eta() - particle1.eta(),
                              particle1.pt(),
-                             particle2.pt());
+                             particle2.pt(), poolBin);
 
       } // end inner loop
     }   // end outer loop
     registry.fill(HIST("hcountDplusHadronPerEvent"), counterDplusHadron);
+    registry.fill(HIST("hZvtx"), mccollision.posZ());
+    registry.fill(HIST("hMultiplicity"), getTracksSize(mccollision));
   }
   PROCESS_SWITCH(HfCorrelatorDplusHadrons, processMcGen, "Process MC Gen mode", false);
+
+  // Event Mixing for the Data Mode
+  using myCollisions = soa::Join<aod::Collisions, aod::Mults, aod::DmesonSelection>;
+  Filter collisionFilter = aod::hf_selection_dmeson_collision::dmesonsel > 0;
+  using mySelCollisions = soa::Filtered<myCollisions>;
+
+  Filter trackFilter = (aod::track::eta > static_cast<float>(-etaTrackMax)) && (aod::track::eta < static_cast<float>(etaTrackMax)) && (aod::track::pt > static_cast<float>(ptTrackMin)) && (aod::track::dcaXY > static_cast<float>(-dcaXYTrackMax)) && (aod::track::dcaXY < static_cast<float>(dcaXYTrackMax)) &&
+                       (aod::track::dcaZ > static_cast<float>(-dcaZTrackMax)) && (aod::track::dcaZ < static_cast<float>(dcaZTrackMax));
+
+  using fullTracks = soa::Join<aod::Tracks, aod::TracksDCA>;
+  using myTracks = soa::Filtered<fullTracks>;
+
+  Filter dplusfilter = aod::hf_sel_candidate_dplus::isSelDplusToPiKPi >= 1;
+  using myCandidatesData = soa::Filtered<soa::Join<aod::HfCand3Prong, aod::HfSelDplusToPiKPi>>;
```

Comment:
Fix the order of struct members.

Reply:
What does this mean @vkucera? I didn't understand it clearly.  I already fixed the other comments.

Reply:
The members should be grouped and sorted. Please see the HF guidelines.

Reply:
done following the guideline given in documentation.